packages/analytics-adapter/src/awaiter.ts

```diff
@@ -0,0 +1,80 @@
+
+export abstract class Awaiter {
+    public interval: number = 50;
+
+    constructor(public readonly maxWaitTime = 5000) {
+        /*
+        In the extending class, call the following in the constructor
+         this.waitForAnalytics();
+         this.waitForEvolv();
+        */
+    }
+
+    abstract getAnalytics(): any;
+    abstract checkAnalyticsProviders(): void; // Optional check for incorrect configuration
+
+    abstract onAnalyticsFound(analytics: any): void;
+    abstract onEvolvFound(): void;
+
+    // Override for customer analytics processor
+    getHandler(): any {
+        return this.getAnalytics();
+    }
```

Comment:
Are both `getHandler()` and `getAnalytics()` needed if `getHandler()` just points to `getAnalytics()`?

Reply:
I need to consider this one further -- getHandler was created for cases where the handler that emitted the event differed from the analytics object we needed to ensure was ready.

This hasn't been needed so far -- but now it might be. getAnalytics for the current integrations is actually the method that fires events back to the analytics tool. Now we're going to be sending back events, but also reading the data layer, we'll probably need to break these apart.

Reply:
Next commit addresses this -- getAnalytics returns the analytics object and is shared. getHandler returns the function to call the events - and is only in the base adapter